src/Compilers/CSharp/Test/Symbol/Symbols/CovariantReturnTests.cs

```diff
@@ -4436,5 +4456,305 @@ void verify(CSharpCompilation compilation)
+            bool withCovariantReturns
+            )
+        {
+            var source = @"
+class Container<T>
+{
+    class Base
+    {
+        public virtual void M(int x) => throw null; // 1
+        public virtual void M(int y) => throw null; // 2
+    }
+
+    class Derived : Base
+    {
+        public override void M(int z) => throw null; // 3
+    }
+}
+";
+            var comp = CreateCompilation(withCovariantReturns: withCovariantReturns, source);
+            comp.VerifyDiagnostics(
+                // (7,29): error CS0111: Type 'Container<T>.Base' already defines a member called 'M' with the same parameter types
+                //         public virtual void M(int y) => throw null; // 2
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "M").WithArguments("M", "Container<T>.Base").WithLocation(7, 29)
+                );
+
+            var member = (SourceMethodSymbol)comp.GlobalNamespace.GetMember("Container.Derived.M");
+            bool useMethodImpl = member.RequiresExplicitOverride(out bool shouldWarn);
+            Assert.False(shouldWarn);
+            Assert.Equal(withCovariantReturns, useMethodImpl);
+
+            verify(SourceView(comp, ""));
+            verify(CompilationReferenceView(comp, ""));
+            verify(RetargetingView(comp, ""));
+
+            void verify(CSharpCompilation compilation)
+            {
+                VerifyOverride(compilation,
+                    methodName: "Container.Derived.M",
+                    overridingMemberDisplay: "void Container<T>.Derived.M(System.Int32 z)",
+                    overriddenMemberDisplay: "void Container<T>.Base.M(System.Int32 x)",
+                    requiresMethodimpl: useMethodImpl);
+            }
+        }
+
+        [Theory]
+        [CombinatorialData]
+        public void DuplicateDeclarations_04(
+            bool withCovariantReturns
+            )
+        {
+            var s0 = @"
+public class Container<T>
+{
+    public class Base
+    {
+        public virtual void M(int x) => throw null; // 1
+        public virtual void M(int y) => throw null; // 2
+    }
+}
+";
+            var baseCompilation = CreateCompilation(withCovariantReturns: withCovariantReturns, s0);
+            baseCompilation.VerifyDiagnostics(
+                // (7,29): error CS0111: Type 'Container<T>.Base' already defines a member called 'M' with the same parameter types
+                //         public virtual void M(int y) => throw null; // 2
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "M").WithArguments("M", "Container<T>.Base").WithLocation(7, 29)
+                );
+            var baseMetadata = baseCompilation.ToMetadataReference();
+
+            var source = @"
+class Derived : Container<int>.Base
+{
+    public override void M(int z) => throw null; // 3
+}
+";
+            var comp = CreateCompilation(withCovariantReturns: withCovariantReturns, source, references: new[] { baseMetadata });
```

Comment:
>comp [](start = 16, length = 4)

I believe this doesn't test retargeting scenario, please add similar test for retargeting scenario.

Reply:
> I believe this doesn't test retargeting scenario, please add similar test for retargeting scenario.

I do not believe this was addressed. I am looking for a similar compilation only with retargeting ```baseCompilation```.  I expect us to get the same error for it.

---
In reply to: [452977229](https://github.com/dotnet/roslyn/pull/45849#discussion_r452977229) [](ancestors = 452977229)

Reply:
It looks like all the tests below also do not test diagnostics for retargeting scenario.

---
In reply to: [454068664](https://github.com/dotnet/roslyn/pull/45849#discussion_r454068664) [](ancestors = 454068664,452977229)